var/spack/repos/builtin/packages/kokkos/package.py

```diff
@@ -89,6 +89,7 @@ class Kokkos(CMakePackage, CudaPackage, ROCmPackage):
         "hpx_async_dispatch": [False, "Whether HPX supports asynchronous dispath"],
         "tuning": [False, "Create bindings for tuning tools"],
         "tests": [False, "Build for tests"],
+        "compile_as_cmake_language": [False, "Use CMake language support for CUDA/HIP"],
```

Comment:
Not sure if it is a good idea to introduce a new flag for old Kokkos version. Perhaps we can limit to 4.2: as we will not be able to test on all old versions.

Reply:
added condition